### PR TITLE
feat(campaign): verify accounts and chain list are associated with an existing campaign id

### LINKED ISSUE #295: campaign(genesis): add a verification all `campaignChains` must be associated to a campaign
Add a verification in the genesis so that all `campaignChains` instances in the genesis are associated to an existing campaign ID

Require #294 and #291 to be merged

### PR DESCRIPTION
**Pull request by Pantani**
Closes #295

### What does this PR does?

add verifications that types must be associated with an existing campaign.

#### How to test?

```shell
go test -v -run TestGenesisState_Validate ./x/campaign/types
```

### CODE REVIEW

File: x/campaign/types/genesis.go [hunk 1]
```diff
@@ -50,6 +53,10 @@ func (gs GenesisState) Validate() error {
 	// Check for duplicated index in mainnetAccount
 	mainnetAccountIndexMap := make(map[string]struct{})
 	for _, elem := range gs.MainnetAccountList {
+		if _, ok := campaignIDMap[elem.CampaignID]; !ok {
+			return fmt.Errorf("campaign id %d not exist for mainnet account %s",
```
Comment (lubtd) [inline x/campaign/types/genesis.go:57]:
```suggestion
			return fmt.Errorf("campaign id %d doesn't exist for mainnet account %s",
```

File: x/campaign/types/genesis.go [hunk 2]
```diff
@@ -40,6 +40,9 @@ func (gs GenesisState) Validate() error {
 	// Check for duplicated index in campaignChains
 	campaignChainsIndexMap := make(map[string]struct{})
 	for _, elem := range gs.CampaignChainsList {
+		if _, ok := campaignIDMap[elem.CampaignID]; !ok {
+			return fmt.Errorf("campaign id %d not exist for chains", elem.CampaignID)
```
Comment (lubtd) [inline x/campaign/types/genesis.go:44]:
```suggestion
			return fmt.Errorf("campaign id %d doesn't exist for chains", elem.CampaignID)
```

File: x/campaign/types/genesis.go [hunk 3]
```diff
@@ -60,6 +67,10 @@ func (gs GenesisState) Validate() error {
 	// Check for duplicated index in mainnetVestingAccount
 	mainnetVestingAccountIndexMap := make(map[string]struct{})
 	for _, elem := range gs.MainnetVestingAccountList {
+		if _, ok := campaignIDMap[elem.CampaignID]; !ok {
+			return fmt.Errorf("campaign id %d not exist for mainnet vesting account %s",
```
Comment (lubtd) [inline x/campaign/types/genesis.go:71]:
```suggestion
			return fmt.Errorf("campaign id %d doesn't exist for mainnet vesting account %s",
```

File: x/campaign/types/genesis_test.go [hunk 4]
```diff
@@ -53,6 +53,81 @@ func TestGenesisState_Validate(t *testing.T) {
 			},
 			valid: true,
 		},
+		{
+			desc: "not exist campaign for mainnet vesting account",
```
Comment (lubtd) [inline x/campaign/types/genesis_test.go:57]:
```suggestion
			desc: "non existing campaign for mainnet vesting account",
```

File: x/campaign/types/genesis_test.go [hunk 5]
```diff
@@ -53,6 +53,81 @@ func TestGenesisState_Validate(t *testing.T) {
...
+					{
+						CampaignID: 9999,
+						Address:    "9999",
+					},
+				},
+			},
+			valid: false,
+		},
+		{
+			desc: "not exist campaign for mainnet account",
```
Comment (lubtd) [inline x/campaign/types/genesis_test.go:90]:
```suggestion
			desc: "non existing campaign for mainnet account",
```

File: x/campaign/types/genesis_test.go [hunk 6]
```diff
@@ -53,6 +53,81 @@ func TestGenesisState_Validate(t *testing.T) {
...
+				CampaignCount: 2,
+				MainnetAccountList: []types.MainnetAccount{
+					sample.MainnetAccount(330, sample.AccAddress()),
+					sample.MainnetAccount(434, sample.AccAddress()),
+				},
+			},
+			valid: false,
+		},
+		{
+			desc: "not exist campaign for chains",
```
Comment (lubtd) [inline x/campaign/types/genesis_test.go:113]:
```suggestion
			desc: "non existing campaign for chains",
```

File: x/campaign/types/genesis_test.go [hunk 7]
```diff
@@ -53,6 +53,81 @@ func TestGenesisState_Validate(t *testing.T) {
...
+					},
+					{
+						CampaignID: 4,
+					},
+				},
+				CampaignList: []types.Campaign{
+					sample.Campaign(99),
+					sample.Campaign(88),
+				},
+				CampaignCount: 2,
```
Comment (lubtd) [inline x/campaign/types/genesis_test.go:127]:
```suggestion
				CampaignCount: 100,
```
Here we break the rule `count > max(id)`

**Review by lubtd — APPROVED** [review]

**Review by ilgooz — APPROVED** [review]

**Pull request merged**